src/sensor_module.c

```diff
@@ -0,0 +1,233 @@
+    {
+        case NMEA_MESSAGE_GGA:
+            {
+                setPacket(&sensor_packet);
+
+                sensor_packet.payload.fixType = NMEA_Data.GGA.fixQuality;
+                type_set = 1;
+
+                if(NMEA_Data.GGA.fixQuality != 0)
+                {
+                    sensor_packet.payload.latitude = (int32_t) (NMEA_Data.GGA.latitude * 1e7);
+                    sensor_packet.payload.longitude = (int32_t) (NMEA_Data.GGA.longitude * 1e7);
+                    sensor_packet.payload.altitude = (uint16_t) (NMEA_Data.GGA.altitude * 10);
+                    location_rdy = 1;
+                    alt_rdy = 1;
+
+                    if(ZDA_ymd_set == 1)
+                    {
+                        sensor_packet.payload.time = changeTime(&tmp_year, &tmp_month, &tmp_day, NMEA_Data.GGA.fixTime);
+                        ymd_ready = 1;
+                    }
+                }
+            }
+            break;
+        case NMEA_MESSAGE_GLL:
+            {
+                setPacket(&sensor_packet);
+
+                if(NMEA_Data.GLL.fixType != 'V')
+                {
+                    sensor_packet.payload.latitude = (int32_t) (NMEA_Data.GLL.latitude * 1e7);
+                    sensor_packet.payload.longitude = (int32_t) (NMEA_Data.GLL.longitude * 1e7);
+                    location_rdy = 1;
+
+                    if(ZDA_ymd_set == 1)
+                    {
+                        sensor_packet.payload.time = changeTime(&tmp_year, &tmp_month, &tmp_day, NMEA_Data.GLL.fixTime);
+                        ymd_ready = 1;
+                    }
+                }
+                else
+                {
+                    sensor_packet.payload.fixType = 0;
+                    type_set = 1;
+                }
+            }
+            break;
+        case NMEA_MESSAGE_ZDA:
+            {
+                setPacket(&sensor_packet);
+
+                if(NMEA_Data.ZDA.fixTime[0] != 0xFF && NMEA_Data.ZDA.fixTime[1] != 0xFF && NMEA_Data.ZDA.fixTime[2] != 0xFF) //ZDA has no fixType like input
+                {
+                    sensor_packet.payload.time = changeTime(&NMEA_Data.ZDA.year, &NMEA_Data.ZDA.month, &NMEA_Data.ZDA.day, NMEA_Data.ZDA.fixTime);
+                    tmp_year = NMEA_Data.ZDA.year;
+                    tmp_month = NMEA_Data.ZDA.month;
+                    tmp_day = NMEA_Data.ZDA.day;
+                    ZDA_ymd_set = 1;
+                    ymd_ready = 1;
+                }
+                else
+                {
+                    sensor_packet.payload.fixType = 0;
+                    type_set = 1;
+                }
+            }
+            break;
+        default:
+            break;
+    }
+    sensor_packet.payload.heading = readCompass(); //the number in dataStore.bin (pValue);
+    sensor_packet.payload.voltage = readVoltage(); //the number in voltageSim.bin (pValue);
+    if(ymd_ready == 1 && location_rdy == 1 && alt_rdy == 1 && type_set == 1)
+    {
+        uint8_t buffer[256];
```

Comment:
Do not declare variables partway through a function, please declare this at the top of the function